src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseResource.java

```diff
@@ -221,14 +221,23 @@ public ResponseEntity<ProgrammingExercise> createProgrammingExercise(@RequestBod
         programmingExercise.validateGeneralSettings();
         programmingExercise.validateProgrammingSettings();
         programmingExercise.validateManualFeedbackSettings();
-        auxiliaryRepositoryService.validateAndAddAuxiliaryRepositoriesOfProgrammingExercise(programmingExercise, programmingExercise.getAuxiliaryRepositories());
-        submissionPolicyService.validateSubmissionPolicyCreation(programmingExercise);
 
         ProgrammingLanguageFeature programmingLanguageFeature = programmingLanguageFeatureService.get()
                 .getProgrammingLanguageFeatures(programmingExercise.getProgrammingLanguage());
 
+        // Check if auxiliary repositories are supported
+        if (programmingExercise.getAuxiliaryRepositories().size() > 0 && !programmingLanguageFeature.auxiliaryRepositoriesSupported()) {
```

Comment:
`programmingExercise.getAuxiliaryRepositories()` could be extracted into a local variable